src/managers/tls.py

```diff
@@ -23,24 +26,100 @@
 class TLSManager:
     """Manager for building necessary files for Java TLS auth."""
 
+    SUBJECT_NAME_MAX_LENGTH = 64
+
     def __init__(self, context: Context, workload: KyuubiWorkloadBase):
         self.context = context
         self.workload = workload
 
-    def get_subject(self) -> str:
-        """Get subject name for the unit."""
+    @property
+    def tls_private_key_secret(self) -> Secret:
+        """Lazily initialize the system users Secret object."""
+        secret_id = self.context.config.tls_client_private_key
+        if secret_id is None:
+            raise RuntimeError("TLS private key secret is not configured.")
+        return Secret(model=self.context.model, secret_id=secret_id)
+
+    def tls_private_key_secret_configured(self) -> bool:
```

Comment:
Similar as before, also these set of checks seems more embedded into charm logic than managers